packages/router/router.ts

```diff
@@ -3627,8 +3629,25 @@ async function callLoaderOrAction(
     let abortPromise = new Promise((_, r) => (reject = r));
     onReject = () => reject();
     request.signal.addEventListener("abort", onReject);
+
+    let runHandlerAndMaybeAbortDeferreds = async (
+      handler: ActionFunction | LoaderFunction,
+      args: LoaderFunctionArgs | ActionFunctionArgs
+    ) => {
+      // Since we Promise.race the handler against the request signal, if the
+      // request is aborted we never get access to the returned value from the
+      // handler.  If we get a defer() instance _after_ we've already aborted
+      // the request, we should proxy that cancellation along so the defer()
+      // controller also aborts.
+      let result = await handler(args);
+      if (args.request.signal.aborted && isDeferredData(result)) {
+        result.cancel();
```

Comment:
Cancel `defer()` instances created _after_ the `request` has already been aborted